### PR TITLE
Update reflection tests

### PR DESCRIPTION
**Pull request by ayg**
This also includes adding support for a few properties with special
behavior, like form.action.  It matches whatwg/html#2950, which matches
implementations.

### CODE REVIEW

File: html/dom/reflection.js [hunk 1]
```diff
@@ -776,25 +781,44 @@ ReflectionTests.reflects = function(data, idlName, idlObj, domName, domObj) {
...
+        // Hard-coded special case
         for (var i = 0; i < domTests.length; i++) {
-            if (domExpected[i] === null && !data.isNullable) {
-                // If you follow all the complicated logic here, you'll find that
-                // this will only happen if there's no expected value at all (like
-                // for tabIndex, where the default is too complicated).  So skip
-                // the test.
-                continue;
+            if (domTests[i] === "") {
+                domExpected[i] = domObj.ownerDocument.URL;
```
Comment (annevk) [inline html/dom/reflection.js]: This should be .baseURI, no? Same below.

**Review by annevk — APPROVED** [review]
Didn't spot anything obviously wrong other than action handling.

**Comment by ayg** [comment]
That doesn't look like what the spec says:

> The action IDL attribute must reflect the content attribute of the same name, except that on getting, when the content attribute is missing or its value is the empty string, the element's node document's URL must be returned instead.
> https://html.spec.whatwg.org/#dom-fs-action

Am I misreading it, or is the spec wrong, or is everything okay and I should merge?

**Comment by ayg** [comment]
I updated for discussion in whatwg/html#2950.  I force-pushed so I could change the commit message, but didn't make any changes other than making form.action work the same as .formAction in both places in reflection.js, not just one.

**Comment by w3c-bots** [comment]
# Build PASSED

Started: 2017-08-21 15:51:38
Finished: 2017-08-21 16:03:52



View more information about this build on:

- [WPT PR Status](http://pulls.web-platform-tests.org/build/17632)
- [TravisCI](https://travis-ci.org/w3c/web-platform-tests/builds/266824101)

**Comment by w3c-bots** [comment]
# Build PASSED

Started: 2017-08-21 15:51:38
Finished: 2017-08-21 16:03:52



View more information about this build on:

- [WPT PR Status](http://pulls.web-platform-tests.org/build/17632)
- [TravisCI](https://travis-ci.org/w3c/web-platform-tests/builds/266824101)

**Comment by w3c-bots** [comment]
# Build PASSED

Started: 2017-08-21 17:57:27
Finished: 2017-08-21 18:06:12


<h2>Failing Jobs</h2>
<ul>

<li>chrome:unstable</li>

</ul>




View more information about this build on:

- [WPT PR Status](http://pulls.web-platform-tests.org/build/17639)
- [TravisCI](https://travis-ci.org/w3c/web-platform-tests/builds/266894380)

**Comment by annevk** [comment]
I shouldn't have guessed.

**Pull request merged**